packages/utils/src/string/extractLetters/index.ts

```diff
@@ -0,0 +1,29 @@
+interface Options {
+  letters?: boolean;
+  numbers?: boolean;
+  specialCharacters?: boolean;
+  whiteSpace?: boolean;
+}
+
+const createRegex = (options: Options) => {
+  const { letters, numbers, specialCharacters, whiteSpace } = options;
+
+  let pattern = '[^';
+
+  if (letters) pattern += '\\p{L}';
+  if (numbers) pattern += '\\p{N}';
+  if (specialCharacters) pattern += '\\p{S}\\p{P}';
+  if (whiteSpace) pattern += `\\s`;
+
+  pattern += `]`;
+
+  return new RegExp(pattern, 'gu');
+};
```

Comment:
```suggestion
const regexps = {
  letters: '\\p{L}',
  numbers: '\\p{N}',
  specialCharacters: '\\p{S}\\p{P}',
  whiteSpace: `\\s`,
} as const;

const createRegex = (options: Options) => {
  let pattern = '';

  for (const key in options) {
    if (options[key as keyof Options]) {
      pattern += regexps[key as keyof Options];
    }
  }

  return new RegExp(`[^${pattern}]`, 'gu');
};
```
1. 정규표현식을 네이밍을 붙여 관리하면 좋을 것 같습니다 🤗
2. 조건식을 여러 번 활용하기 보다 for문을 활용하는건 어떨까요!?